octoprint_mrbeam/filemanager/__init__.py

```diff
@@ -108,18 +112,24 @@ def _delete_files_by_age(self, files, num_files_to_keep=0):
     def _is_history_file(entry):
         _, extension = os.path.splitext(entry)
         extension = extension[1:].lower()
-        return extension == "mrb"
+        return extension in MrbFileManager.FILE_EXTENSIONS_HISTORY
 
     @staticmethod
     def _is_gcode_file(entry):
         _, extension = os.path.splitext(entry)
         extension = extension[1:].lower()
-        return extension == "gco"
+        return extension in MrbFileManager.FILE_EXTENSIONS_GCODE
 
     @staticmethod
     def _sanitize_content(file_name, content):
         _, extension = os.path.splitext(file_name)
-        if extension in (".svg", ".mrb", ".g", ".gco", ".gc", ".gcode", ".nc"):
+        extension = extension[1:].lower()
+        if (
+            extension
+            in MrbFileManager.FILE_EXTENSIONS_SVG
+            # + MrbFileManager.FILE_EXTENSIONS_GCODE
+            # + MrbFileManager.FILE_EXTENSIONS_HISTORY
+        ):
             # TODO stripping non-ascii is a hack - svg contains lots of non-ascii in <text> tags. Fix this!
             content = "".join(i for i in content if ord(i) < 128)
         return content
```

Comment:
I know it's no the purpose of the PR, but shouldn't the string be converted to a unicode string instead? Or replace the characters with something else so it's easier to diagnose (such as a "?").

I would mark this with a comment for the py3 transition - with unicode support backend into the language, this step would be redundant. You can also use `sys.version_info < (3, 0, 0)` to skip this step on python3.